Source/Rio.EFModels/Entities/ParcelLedgers.cs

```diff
@@ -143,14 +144,39 @@ public static ParcelLedgerDto GetByIDAsDto(RioDbContext dbContext, int parcelLed
             return parcelLedger?.AsDto();
         }
 
-        public static int CreateNew(RioDbContext dbContext, ParcelLedgerCreateDto parcelLedgerCreateDto, int userID)
+        public static void CreateNew(RioDbContext dbContext, ParcelDto parcel, ParcelLedgerCreateDto parcelLedgerCreateDto, int userID)
+        {
+            foreach (var parcelNumber in parcelLedgerCreateDto.ParcelNumbers)
+            {
+                var parcelLedger = new ParcelLedger()
+                {
+                        ParcelID = parcel.ParcelID,
+                        TransactionDate = DateTime.UtcNow,
+                        EffectiveDate = parcelLedgerCreateDto.EffectiveDate,
+                        TransactionTypeID = parcelLedgerCreateDto.TransactionTypeID,
+                        ParcelLedgerEntrySourceTypeID = (int)ParcelLedgerEntrySourceTypeEnum.Manual,
+                        TransactionAmount = parcelLedgerCreateDto.TransactionAmount,
+                        WaterTypeID = parcelLedgerCreateDto.WaterTypeID,
+                        TransactionDescription =
+                            $"A manual {(parcelLedgerCreateDto.TransactionAmount < 0 ? "withdrawal from" : "deposit to")} water {(parcelLedgerCreateDto.WaterTypeID.HasValue ? "supply" : "usage")} has been applied to this water account.",
+                        UserID = userID,
+                        UserComment = parcelLedgerCreateDto.UserComment
+                };
+                dbContext.ParcelLedgers.Add(parcelLedger);
+            }
+            dbContext.SaveChanges();
+        }
+
+        public static int BulkCreateNew(RioDbContext dbContext, ParcelLedgerCreateDto parcelLedgerCreateDto, int userID)
         {
             int createdCount = 0;
             foreach (var parcelNumber in parcelLedgerCreateDto.ParcelNumbers)
             {
+                var parcel = dbContext.Parcels.SingleOrDefault(x => x.ParcelNumber == parcelNumber);

```

Comment:
I think you want to make this call once to grab all the parcels for all the parcelnumbers instead of doing it one at a time

Reply:
Just created a Parcel.ListByParcelNumbers() function & looping through the returned list of parcels instead of parcelLedgerCreateDto.ParcelNumbers